update: internal visibility description (#5685)
JetBrains/kotlin-web-site: docs/topics/jvm/java-to-kotlin-interop.md

> The Kotlin visibility modifiers map to Java in the following way:

Reviewer: (suggestion)
What do you think about rewriting this sentence slightly? 

i.e.: The Kotlin compiler maps visibility modifiers to Java in the following way:

That way we could turn the sentences active easily. 

"`private` members remain `private` members.
It compiles private top-level declarations to private -- etc.

* `private` members are compiled to `private` members.
* `private` top-level declarations are compiled to `private` top-level declarations. Package-private accessors are also
  included if accessed from within a class. 
* `protected` members remain `protected`.

  Note that Java allows accessing protected members from other classes in the same package and Kotlin doesn't,

>   so Java classes have broader access to the code.

Reviewer: I'm not sure if this last clause adds anything, but up to you. Imho we could ommit it. To me it feels clear based on the description that this is the case.

* `internal` declarations become `public` in Java.

  Members marked as `internal` go through name mangling in bytecode. This prevents accidental overrides across modules,
  for example, when extending a Kotlin class from Java, and allows overloading for members with the same signature that

>   don't see each other according to Kotlin rules.

Reviewer: I know this was in the original, but "don't see each other" is a difficult sentence to parse because of the anthropomorphism.

WDYT about this streamlined version? 

> The compiler mangles the names of members marked as `internal` in the bytecode. 
This prevents a member in another module from accidentally overriding an internal member with the same signature.

Author: I'd still keep a bit more detailed explanation with an example, but I agree that the last part can be dropped

Reviewer: I like it! 👍

>   Note that the names of public members of `internal` classes are not mangled and remain callable from Java.

Reviewer: ```suggestion
  Note that the names of public members of `internal` classes aren't mangled and remain callable from Java.
```
